src/main/java/org/adoptopenjdk/modulescanner/Main.java

```diff
@@ -19,25 +23,48 @@
 
     private static String DEFAULT_DIRECTORY_TO_SCAN = "../gs-maven-mirror";
     private static String CUTOFF_DATE = "20170101000000";
+    private static String DEFAULT_OUTPUT_FILE_NAME = "modulescanner.csv";
 
     /**
      * Main method - entry point for invoking modulescanner
      *
      * @param args Commandline arguments
      */
-    public static void main(String[] args) {
+    public static void main(String... args) {
         var directoryToScan = Paths.get(args.length > 0 ? args[0] : DEFAULT_DIRECTORY_TO_SCAN);
         var cutoffDate = args.length > 1 ? args[1] : CUTOFF_DATE;
+        var output = Paths.get(args.length > 2 ? args[2] : DEFAULT_OUTPUT_FILE_NAME);
+
+        try (var out = new PrintWriter(new BufferedWriter(new FileWriter(output.toFile())))) {
+            walk(directoryToScan, cutoffDate, out);
+            LOGGER.info("Wrote " + Files.size(output) + " bytes to: " + output);
+        } catch (IOException ioe) {
+            LOGGER.error("Creating output file " + output + " failed", ioe);
+        }
+    }
+
+    // Walk repository and emit CSV file as output
+    private static void walk(Path directoryToScan, String cutoffDate, PrintWriter out) {
+        var csv = new CsvPrinter(out, ", ");
```

Comment:
Spaces are significant in csv, so better leave the trailing space off.

Reply:
Sure. Or switch to tabs as delimiter? Without spaces `"\t"` and renaming the helper class to `SeparatedValuesPrinter`?